WorkflowSwiftUIExperimental/Sources/SwiftUIScreen.swift

```diff
@@ -77,17 +95,104 @@ private struct EnvironmentInjectingView<Content: View>: View {
 private final class ModeledHostingController<Model, Content: View>: UIHostingController<Content> {
     let modelSink: Sink<Model>
     let viewEnvironmentSink: Sink<ViewEnvironment>
+    var swiftUIScreenSizingOptions: SwiftUIScreenSizingOptions {
+        didSet {
+            updateSizingOptionsIfNeeded()
 
-    init(modelSink: Sink<Model>, viewEnvironmentSink: Sink<ViewEnvironment>, rootView: Content) {
+            if !hasLaidOutOnce {
+                setNeedsLayoutBeforeFirstLayoutIfNeeded()
+            }
+        }
+    }
+
+    private var hasLaidOutOnce = false
+
+    init(
+        modelSink: Sink<Model>,
+        viewEnvironmentSink: Sink<ViewEnvironment>,
+        rootView: Content,
+        swiftUIScreenSizingOptions: SwiftUIScreenSizingOptions
+    ) {
         self.modelSink = modelSink
         self.viewEnvironmentSink = viewEnvironmentSink
+        self.swiftUIScreenSizingOptions = swiftUIScreenSizingOptions
 
         super.init(rootView: rootView)
+
+        updateSizingOptionsIfNeeded()
     }
 
     required init?(coder aDecoder: NSCoder) {
         fatalError("not implemented")
     }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = .clear
+
+        setNeedsLayoutBeforeFirstLayoutIfNeeded()
+    }
+
+    override func viewDidLayoutSubviews() {
```

Comment:
> Market also supports having the preferredContentSize automatically calculated if it's requested by a consumer via an associated value on UIViewController—we'll likely want to figure out a way to support this with SwiftUIScreens, and that may involve moving some of that infra into the Workflow repo ([UI-5801](https://block.atlassian.net/browse/UI-5801)).

Hmm yeah this is an interesting one, since that stuff all lives in Modals right now. We'll definitely want to keep it, since avoiding the duplicative preferredContentSize calculations was a pretty big performance win.

Trying to think if there's a way we could flow this into the SwiftUI environment at the Market layer... But I guess since this is a view controller, we don't have that available.

One gross idea, but it would work, is do a `respondsToSelector` check on the associated value we added, and reference it via `valueForKey` in this repo. Icky but it would do the thing we want...

Reply:
I think we may want to consider moving that associated object to a `ViewEnvironmentKey`, and then move that key's definition to `ViewEnvironmentUI`, possibly also adding support for this automatically in `WorkflowUI` (along with here).

Let's move further discussion on this to [UI-5801](https://block.atlassian.net/browse/UI-5801) so that the context is easier to discover once we start that work!